### ISSUE TITLE
scipy.stats.mstats.theilslopes breks for consecutive similar values

### ISSUE THREAD

**Issue by tammasloughran** [scipy/scipy#5973]
scipy.stats.mstats.theilslopes calculates the slope and intercept for a set of data by finding the median of the slopes between all points. If the data has consecutive elements that have the same value, it doesn't work. The slope between these points would be zero, but I don't see any reason why that would cause problems. It seems to happen at four or more consecutive values. 

import numpy as np
from scipy.stats.mstats import theilslopes
theilslopes(np.array([1.3,2.5,1.9,3,1.12,1.1]))
theilslopes(np.array([1.3,2.5,1.9,3,1.12,1.1,1.1,1.1]))
theilslopes(np.array([1.3,2.5,1.9,3,1.12,1.1,1.1,1.1,1.1]))

/usr/lib/python2.7/dist-packages/scipy/stats/mstats_basic.py:703: RuntimeWarning: invalid value encountered in sqrt
  sigma = np.sqrt(sigsq)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/scipy/stats/mstats_basic.py", line 707, in theilslopes
    delta = slopes[[Rl,Ru]]
  File "/usr/local/lib/python2.7/dist-packages/numpy/ma/core.py", line 3111, in __getitem__
    dout = self.data[indx]
IndexError: index -9223372036854775808 is out of bounds for axis 1 with size 36

Is this normal? I'm using scipy version 0.9.0

**Comment by ev-br** [comment]
Thanks for the report. This seems to have been fixed in some more recent versions of scipy (0.9.0 is *really* worth upgrading from)

```
In [1]: from scipy.stats import theilslopes

In [2]: import numpy as np

In [3]: theilslopes(np.array([1.3,2.5,1.9,3,1.12,1.1,1.1,1.1,1.1]))
Out[3]: (-0.03666666666666666, 1.2666666666666668, -0.34999999999999998, 0.0)

In [4]: from scipy.stats.mstats import theilslopes as m_theilslopes

In [5]: m_theilslopes(np.array([1.3,2.5,1.9,3,1.12,1.1,1.1,1.1,1.1]))
Out[5]: (-0.03666666666666666, 1.2666666666666668, -0.34999999999999998, 0.0)

In [6]: import scipy

In [7]: scipy.__version__
Out[7]: '0.18.0.dev0+8b07439'
```

**Comment by kokorev** [comment]
I encounter this problem with scipy 0.18.1 but I think it is only a problem when all values are the same

```
from scipy.stats import theilslopes
y = range(10)
x = [1] * 10
r = theilslopes(y, x)
```

produce 
```
r = theilslopes(y, x)
  File ".../python3.5/site-packages/scipy/stats/_stats_mstats_common.py", line 219, in theilslopes
    delta = slopes[[Rl, Ru]]
IndexError: index 0 is out of bounds for axis 1 with size 0
```
